helm/templates/deployment.yaml

```diff
@@ -46,6 +46,10 @@ spec:
             value: {{ .Values.prom2teams.connector | quote }}
           - name: PROM2TEAMS_GROUP_ALERTS_BY
             value: {{ .Values.prom2teams.group_alerts_by | quote }}
+          {{- if .Values.servicemonitor.enabled }}
```

Comment:
I guess this condictional should not be needed, we could just have a value like: 
`value: {{ .Values.prom2teams.prometheus_metrics_enabled | quote }}` as the rest of default environment variables. It could be set in `helm/vars.yml` as `false`

In this way other users that may not be using ServiceMonitor can enable prometheus metrics too.